File: api/src/main/java/gg/mineral/api/world/property/Difficulty.java
package gg.mineral.api.world.property;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.val;

@RequiredArgsConstructor
public enum Difficulty {
    PEACEFUL((byte) 0), EASY((byte) 1), NORMAL((byte) 2), HARD((byte) 3);

    @Getter
    private final byte id;

    public static Difficulty fromId(byte id) {
        for (val difficulty : Difficulty.values())
            if (difficulty.getId() == id)
                return difficulty;

        return null;
    }
}

File: api/src/main/java/gg/mineral/api/world/property/Difficulty.kt
package gg.mineral.api.world.property

import lombok.Getter
import lombok.RequiredArgsConstructor

@RequiredArgsConstructor
enum class Difficulty {
    PEACEFUL(0.toByte()), EASY(1.toByte()), NORMAL(2.toByte()), HARD(3.toByte());

    @Getter
    private val id: Byte = 0

    companion object {
        fun fromId(id: Byte): Difficulty? {
            for (difficulty in entries) if (difficulty.getId() == id) return difficulty

            return null
        }
    }
}
